### PR TITLE
Added remove techniques stage

### PR DESCRIPTION
**Pull request by leerichard42**

### CODE REVIEW

**Comment by pjcozzi** [comment]
@leerichard42 make sure to do a `git pull` from this branch.  I merged in master and fixed the conflicts.

File: lib/removeUnusedTechniques.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+function removeUnusedTechniques(gltf, stats) {
+    var usedTechniqueIds = {};
+    var materials = gltf.materials;
+
+    // Build hash of used techniques by iterating through materials
+    if (defined(materials)) {
+        for (var materialId in materials) {
+            if (materials.hasOwnProperty(materialId)) {
+                var id = materials[materialId].technique;
```
Comment (pjcozzi) [inline lib/removeUnusedTechniques.js]: Careful, `technique` is optional.  We need a `defined` check here.

See https://github.com/KhronosGroup/glTF/blob/master/specification/README.md#materialtechnique

File: lib/removeUnusedTechniques.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        if (defined(stats)) {
+            stats.numberOfTechniquesRemoved += numberOfTechniquesRemoved;
+        }
+
+        gltf.techniques = usedTechniques;
+    }
+
+// TODO: remove orphan uris
```
Comment (pjcozzi) [inline lib/removeUnusedTechniques.js]: Remove this.

**Comment by pjcozzi** [comment]
Looks good!  Just those minor comments.

**Comment by leerichard42** [comment]
Updated!

**Comment by pjcozzi** [comment]
Nice! :+1:

**Pull request merged**